### PR TITLE
Keep `GraphNode` port icons crisp at high zoom levels and remove artifacts

### PR DESCRIPTION
**Pull request by Geometror**
With zoom levels up to 207%, the port icons are noticeably stretched. You can also observe interpolation/sampling artifacts because the circle touches all edges of the SVG's viewport. This can be solved with size overrides and adjusting the circle/the image's dimensions.

Before/After:
1QVjFujSBJ AfgNu6m39j

### CODE REVIEW

**Comment by Calinou** [comment]
TIL about `ImageTexture.set_size_override`. This method seems pretty useful and I'm surprised it's not exposed as a property, nor in the inspector for ImageTextures ([it is for PortableCompressedTexture2D though](https://github.com/godotengine/godot/blob/85c9db592f22941ce7c2a540a526e702f4cc1848/scene/resources/texture.cpp#L628)).

PS: The class reference description for `ImageTexture.set_size_override()` is misleading, as it overrides the size the texture is rendered at (only affects 2D). It does not actually resize the image data.

**Comment by Geometror** [comment]
Same as you, I just learned about that method today and wished I had sooner. (I stumbled upon it accidentally while looking for a way to implement this exact functionality)
I think we should definitely expose `size_override` as a property or increase its visibility in another way, at least through the documentation.
When even you, Calinou, didn't know about it, then we definitely need to take some action :)

File: editor/editor_themes.cpp [hunk 1]
```diff
@@ -1910,7 +1910,10 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 
 	theme->set_icon("close", "GraphNode", theme->get_icon(SNAME("GuiCloseCustomizable"), SNAME("EditorIcons")));
 	theme->set_icon("resizer", "GraphNode", theme->get_icon(SNAME("GuiResizer"), SNAME("EditorIcons")));
-	theme->set_icon("port", "GraphNode", theme->get_icon(SNAME("GuiGraphNodePort"), SNAME("EditorIcons")));
+	Ref<ImageTexture> port_icon = theme->get_icon(SNAME("GuiGraphNodePort"), SNAME("EditorIcons"));
+	// The true size is 20x20. This is necessary for sharp port icons at high zoom levels in GraphEdit (up to 200%).
```
Comment (YuriSizov) [inline editor/editor_themes.cpp:1914]: 24 by 24 actually.

File: editor/icons/GuiGraphNodePort.svg [hunk 2]
```diff
@@ -1 +1 @@
-<svg height="10" viewBox="0 0 10 10" width="10" xmlns="http://www.w3.org/2000/svg"><circle cx="5" cy="5" fill="#fff" r="5"/></svg>
+<svg height="24" width="24" xmlns="http://www.w3.org/2000/svg"><circle cx="12" cy="12" fill="#fff" r="10"/></svg>
```
Comment (YuriSizov) [inline editor/icons/GuiGraphNodePort.svg:1]: `viewBox` is still required.

Comment (Geometror) [reply editor/icons/GuiGraphNodePort.svg]: Fixed. Unfortunately, my minification tool (SVG VSCode extension which uses SVGO under the hood) always removes viewBox. What do you use for SVG minification?

Comment (YuriSizov) [reply editor/icons/GuiGraphNodePort.svg]: SVGCleaner, https://github.com/RazrFalcon/svgcleaner. Though SVGO should have configuration options: https://github.com/svg/svgo/issues/505

**Review by YuriSizov — APPROVED** [review]
That's an interesting trick. I wonder if we can utilize it elsewhere.

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**